File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> map1;
        unordered_map<char, char> map2;
        
        for(int i = 0; i < s.length(); i++)
        {
            if(map1.find(s[i]) == map1.end())
            {
                map1[s[i]] = t[i];
            }
            else
            {
                if(map1[s[i]] != t[i]) return false;
                
            }

            if(map2.find(t[i]) == map2.end())
            {
                map2[t[i]] = s[i];
            }
            else
            {
                if(map2[t[i]] != s[i]) return false;
                
            }
        }
        
        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        dic = dict()
        
        for i in range(len(s)):
            
            if s[i] not in dic:
                dic[s[i]] = t[i]
            else:
                if dic[s[i]] != t[i]:
                    return False
                
                
        dic = dict()
        
        for i in range(len(s)):
            
            if t[i] not in dic:
                dic[t[i]] = s[i]
            else:
                if dic[t[i]] != s[i]:
                    return False
                
        
        return True